src/cpp/security/accesscontrol/DistinguishedName.h

```diff
@@ -0,0 +1,201 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file DistinguishedName.h
+ */
+
+#ifndef _SECURITY_ACCESSCONTROL_DISTINGUISHEDNAME_H_
+#define _SECURITY_ACCESSCONTROL_DISTINGUISHEDNAME_H_
+
+#include <array>
+#include <string>
+
+namespace eprosima {
+namespace fastrtps {
+namespace rtps {
+namespace security {
+
+/*
+ * RFC 1779: https://datatracker.ietf.org/doc/html/rfc1779 specifies how to represent DistinguishedNames
+ * in string format (obsolete in favor of rfc2253).
+ *
+ * RFC 2253: https://datatracker.ietf.org/doc/html/rfc2253 specifies how to represent DistinguishedNames
+ * in string UTF-8 format.
+ *
+ * ASN.1 representation of DistinguishedNames
+ * DistinguishedName ::= RDNSequence
+ * RDNSequence ::= SEQUENCE OF RelativeDistinguishedName
+ * RelativeDistinguishedName ::= SET SIZE (1..MAX) OF
+ * AttributeTypeAndValue
+ * AttributeTypeAndValue ::= SEQUENCE {
+ * type  AttributeType,
+ * value AttributeValue }
+ *
+ * Each RDN in a RDNSequence is separated by ','.
+ * Each AttributeTypeAndValue in a RDN is separated by '+'.
+ * Attribute and Value are separated by '='.
+ *
+ */
+
+/**
+ * @brief class that represents a Distinguished Name
+ *
+ * @note so far this implementation works only as a string
+ * @todo make this a real class with real methods that correctly implement RFC 2253:
+ */
+using DistinguishedName = std::string;
+
+/**
+ * @brief This function compares 2 strings representing DistinguishedNames.
+ *
+ * Follow RFC 2253: https://datatracker.ietf.org/doc/html/rfc2253
+ * The idea here is that comparing 2 DistinguishedNames is not as trivial as comparing 2 strings,
+ * nor comparing its values splitting the string.
+ * The comparison must be a complex function that takes into account many specific details.
+ *
+ * So far, Fast DDS gives 2 strings (in some format that we assume is the correct one) and this compares
+ * whether the two strings refer to the same DistinguishedName.
+ * It would be better (and easier) to store DistinguishedName in a specialized function instead of in a string,
+ * and thus this comparison would be much easier and efficient.
+ * But this will be left for future improvements.
+ *
+ * @warning this function does not correctly fulfilled all requirements of RFC 2253.
+ * Requirements that are not fulfilled:
+ * - multi-attribute
+ * - types accept oid
+ * - assumes that input is a well formed DistinguishedName following rfc2253
+ * @todo make this function complete (probably better by implementing DistinguishedName class).
+ *
+ * @param name1 first DistinguishedName in string format
+ * @param name2 second DistinguishedName in string format
+ *
+ * @return true if both first strings represent the same DistinguishedName.
+ * @return false otherwise
+ */
+bool rfc2253_string_compare(
+        const DistinguishedName& name1,
+        const DistinguishedName& name2);
+
+
+namespace details {
+
```

Comment:
A very brief description of what is the `Attribute` struct would be nice (If someone wants to investigate further  you already left the url to the reference document)